kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp

```diff
@@ -237,23 +237,97 @@ namespace Kratos
+            NodeType::Pointer p_node_6 = this_model_part.CreateNewNode(6, 2.0 , 1.0 , 0.0);
+
+            // Now we create the "conditions"
+            std::vector<NodeType::Pointer> element_nodes_0 (3);
+            element_nodes_0[0] = p_node_1;
+            element_nodes_0[1] = p_node_2;
+            element_nodes_0[2] = p_node_3;
+            Triangle2D3 <NodeType>::Pointer p_triangle_0 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_0} );
+
+            std::vector<NodeType::Pointer> element_nodes_1 (3);
+            element_nodes_1[0] = p_node_1;
+            element_nodes_1[1] = p_node_3;
+            element_nodes_1[2] = p_node_4;
+            Triangle2D3 <NodeType>::Pointer p_triangle_1 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_1} );
+
+            std::vector<NodeType::Pointer> element_nodes_2 (3);
+            element_nodes_2[0] = p_node_2;
+            element_nodes_2[1] = p_node_5;
+            element_nodes_2[2] = p_node_3;
+            Triangle2D3 <NodeType>::Pointer p_triangle_2 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_2} );
+
+            std::vector<NodeType::Pointer> element_nodes_3 (3);
+            element_nodes_3[0] = p_node_5;
+            element_nodes_3[1] = p_node_6;
+            element_nodes_3[2] = p_node_3;
+            Triangle2D3 <NodeType>::Pointer p_triangle_3 = Kratos::make_shared<Triangle2D3 <NodeType>>( PointerVector<NodeType>{element_nodes_3} );
+
+            Element::Pointer p_elem_0 = Kratos::make_shared<Element>(1, p_triangle_0, p_elem_prop);
+            Element::Pointer p_elem_1 = Kratos::make_shared<Element>(2, p_triangle_1, p_elem_prop);
+            Element::Pointer p_elem_2 = Kratos::make_shared<Element>(3, p_triangle_2, p_elem_prop);
+            Element::Pointer p_elem_3 = Kratos::make_shared<Element>(4, p_triangle_3, p_elem_prop);
+            this_model_part.AddElement(p_elem_0);
+            this_model_part.AddElement(p_elem_1);
+            this_sub_model_part.AddElement(p_elem_2);
+            this_sub_model_part.AddElement(p_elem_3);
+
+            // Compute process
+            Parameters settings( R"(
+            {
+                "element_name"        : "Element2D3N",
+                "condition_name"      : "",
+                "use_root_model_part" : false
+
+            }  )" );
+
+            ReplaceElementsAndConditionsProcess process = ReplaceElementsAndConditionsProcess(this_sub_model_part, settings);
+            process.Execute();
+
+            // Same element type, same geometry type
+            std::string component_name;
+            for (auto& r_element : this_sub_model_part.Elements()) {
+                CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name);
+                KRATOS_CHECK_EQUAL(component_name, "Element2D3N");
+            }
+        }
```

Comment:
I don't know how to test the elements that were not replaced
Unfortuanately the same way of checking does not work if the elements are baseclass-elements 
Also there are no other elements in the core that I could use

Does anyone have an idea?

Reply:
Compare elements is possible, but is a little bit of work, look at the model_part_io.h when the name of the output element is checked it does a loop over the KratosComponents<Element> and KratosComponents<Condition>

Reply:
comparing elements yes, but the elements in the MainModelpart (i.e. the ones that are not replaced) are Baseclass-elements => the `CompareElementsAndConditionsUtility` fails here :/